### ISSUE TITLE
Adding a new game type package

### ISSUE THREAD

**Issue by ramzes222** [NPBruce/valkyrie#622]
Hi!
     I'm trying to make a campaign using Valkyrie for Dark Souls the Board Game.
I know this is not a game based upon scenarios, however I would like to make a campaign that would add exploration, lore and meaningful character progression to the board game. I've already been developing the campaign for a few weeks, so I began to think about using Valkyrie to actually play the scenario.
     I have messaged NPBruce and asked him about this. He said that it is possible, I just need to make a content pack for MoM (easier) or a package system for the new game type (harder, but better). I don't need any advanced features, only the basic things, such as tiles, tokens and NPC dialogue. However, I would like to avoid using MoM editor, as there is no way of disabling investigator selection or Mythos Phase. That's why I would prefer option number 2, creating a new game package.
     I thought it would be good to write about it here, as maybe somebody has already done something like this or knows how to do this. 

I would really appreciate any help :)

**Comment by NPBruce** [comment]
To clarify:

There is no reason not to use the editor.  Investigator/Hero selection is a part of the game type and manually creating scenarios will not avoid this.  For now you can just ignore it and get started.

All game assets can be created easily enough with a content pack that can build on top of MoM/D2E.  As I am not familiar with Dark Souls.  I don't know what would be most appropriate settings.  For reference it goes something like this:

| Game                       | D2E         | MoM                         | DS  |
|----------------------------|-------------|-----------------------------|-----|
| Layout                     | Grid        | Freeform                    | ?   |
| Activation                 | Alternating | Investigators then Monsters | ?   |
| Hero Selection             | Yes         | Yes                         | No? |
| Hero Selection during Game | Yes         | No                          | ?   |
| Monsters Grouped           | Yes         | No                          | ?   |
| Items Selectable           | No          | Yes                         | ?   |
| Starting Items             | No          | Yes                         | ?   |

There are also things like what fonts to use.

**Comment by ramzes222** [comment]
I'll try to explain what I would like to do in the Dark Souls Scenario

**Layout**: Tiles in DS are all the same size and shape (square) and are layed in a grid. On the tiles themselves there are nodes, which can be used to show where exactly on a tile something is. Here's an example of how it looks like:
https://static1.squarespace.com/static/56728f72a128e6b1e548ec55/t/589dfe099de4bba912b5f78f/1486749226351/DS-Boards-AnorLondo.png

**Activation**: In the game the turns are Player1->All Monsters->Player2->All Monsters etc., but I don't think this needs to be in the scenario. The beaviour of the monsters is shown on certain cards, allowing players to make strategic decisions as they know what the enemy will do. I don't think I want to change that so I don't think that the phases should be in the app, as I can think of no way I (or anybody else making a campaign for DS) would use it

**Hero Selection**: Yes. There are 10 classes and which classes were chosen impacts some small parts of the story and the starting items.

**Hero Selection during Game**: No

**Monsters**: In DS, the enemies appear during encounters. Each encounter is a separate tile, and ends with either the players or the monsters dead. That's why there is no need of keeping track of enemies in the app. However, every time a player enters a new room the app should show a description of the place, and, in most cases, the placement and types of enemies in that room. I think the best way of doing this would be sort of like when the players go into a new room in MoM: There is some text and the camera moves to show where you place which tokens (or in case of DS, also monsters).

**Items**: Just like in the original Dark Souls I want to tell the story through NPC dialogue and item descriptions. The first can be done through interactable tokens, but the second one is a bit more tricky. Basically every time the players find an item it appears in the Item menu, where it can be inspected to read it's desription. I have been thinking to do the same thing with enemies with a second menu to give the players more information about the scenario's story. The reason I think it has to be done differently from MoM is that in MoM the players only have a few items, while in DS this ammount can come close to or even over 50.

One more thing: is it possible to show only a part of the map at a time? One box of Dark Souls doesn't have enough tiles to make all of the scenario at once, so I divided it into areas. I think it will help the players a lot if they only see the tiles that should be on the table right now, so they are not confused about this.

I hope I described what I would like to do well. I tried to think not only about my campaign, but also what could be used by other people making a DS scenario in the future. 

Thank you a lot for helping me with this:)

**Comment by NPBruce** [comment]
Almost all of this can be done without changing Valkyrie.

Try putting this in valkyrie_Data\StreamingAssets\content\MoM:
[ds.zip](https://github.com/NPBruce/valkyrie/files/1370056/ds.zip)

You should be able to turn it on and off in content selection and can edit the files to add whatever content you want.

Grid:
The snapping for MoM is a little odd because of the token/board size comparisons, so you might have issues with the editor snap function, but otherwise this should work just fine.  You can manually edit the position in your scenario ini files for now also.

Activation:
Sounds like one room would be one 'investigator phase' and you can progress to the next stage when the room is cleared.  I can't change the investigator phase text easily at the moment, but that can come as a future feature.  No monster activations will be used - this is easy just don't use spawns.

Hero selection:
You can add the classes as 'investigators'  as above some work would be required on my end to rename this.

Monsters:
OK so these will just be tokens, no problems there.

Items:
This can be done currently, but as you say the UI is not designed for 50 items.  As a board game adaptation my first reaction is that this is too much anyway, and should be limited.  When I do a new game type it wouldn't be hard to change the item selection UI, but coming up with a good mobile device interface for this might not be easy anyway.

Showing part of the map:
No problem, you can and and remove tiles whenever you want (in events).  Also note that a scenario in Valkyrie can be linked to other scenarios to form a campaign.  This can be easier to develop and manage.  Also it means that updating the campaign (getting a new version) will take effect when you change scenario, which can be important for bug fixes in long campaigns (though it means you have to maintain compatability on scenario changes).

Before starting on anything my end I would like to see how this goes, as I am sure there will be additional requirements.  I see not reason you can't do all of this now, just that some of the labelling will be wrong.

**Comment by ramzes222** [comment]
I did what you told me to (putting it in MoM folder, but the tiles don't appear. I looked through the tiles, tried to look at how the other content packs did it and change the files according to it, but the tiles still don't appear in the editor.

Btw. I enebled the content pack through "select content"

**Comment by NPBruce** [comment]
Sorry, typo (TileSide not Tile).
[ds.zip](https://github.com/NPBruce/valkyrie/files/1371082/ds.zip)
